alonzo/formal-spec/txinfo.tex

```diff
@@ -3,5 +3,232 @@ \section{TxInfo Construction}
+corresponding Plutus type by $t_P$.
+
+Translating certain kinds of certificates drops the data in the
+certificates, in particular, the $\DCertGen$ and $\DCertMir$ ones.
+
+The constant returned by $\fun{slotToPOSIXTime}$,
+
+\[\type{ResolutionConstant}\]
+
+Is the maximum possible value of a POSIX time interval. This will be adjusted in
+future eras to reflect the time being translated.
+
+Note that the $\Ptr$ addresses translated and passed to Plutus scripts are
+not resolvable to their corresponding key or script staking credencials. This
+is because doing a lookup in the resolution map, which part of the ledger state,
+could break determinism.
+
+During the transaction encoding and decoding process, a transaction is discarded if it is not
+encoded correctly. This includes, in particular, a check that scripts and $\Data$
+terms therein are constructed with constructors of terms of these types, themselves
+applied to correctly typed terms. Functions that perfom these checks are
+in Figure \ref{fig:data-script-check}. The $\fun{P.validateScript}$ function
+is a Plutus-library function which checks whether a bytestring represents a
+Plutus script.
+
+\begin{figure*}[htb]
+  \begin{align*}
+    & \emph{Hash Types} \\
+    & \ScriptHash & \type{P.ValidatorHash} \\
+    & \KeyHash & \type{P.PubKeyHash} \\
+    & \DataHash & \type{P.DatumHash} \\
+    & \TxId & \type{P.TxId} \\
+    ~\\
+    & \emph{Transaction Input, ie. Output Reference}  \\
+    & \TxIn & \type{P.TxOutRef} \\
+    ~\\
+    & \emph{Credential Types} \\
+    & \Ptr & \type{P.StakingPtr} \\
+    & \Credential & \type{P.Credential} \\
+    & \Credential & \type{P.StakingHash} \\
+    & \StakeCredential & \type{P.StakingCredential} \\
+    ~\\
+    & \emph{Credentials and Withdrawals} \\
+    & \Wdrl & \type{P.StakingCredential} \times \Integer \\
+    & \PolicyID & \type{P.CurrencySymbol} \\
+    & \AssetName & \type{P.TokenName} \\
+    ~\\
+    & \emph{Certificate Types} \\
+    & \DCert & \type{P.DCert} \\
+    & \DCertRegKey & \type{P.DCertDelegRegKey} \\
+    & \DCertDeRegKey & \type{P.DCertDelegDeRegKey} \\
+    & \DCertDeleg & \type{P.DCertDelegDelegate} \\
+    & \DCertRetirePool & \type{P.DCertPoolRetire} \\
+    ~\\
+    & \emph{Data-dropping Certificate Types} \\
+    & \DCertGen & \type{P.DCertGenesis} \\
+    & \DCertMir & \type{P.DCertMir} \\
+  \end{align*}
+  \caption{TxInfo and Constituent Types}
+  \label{fig:txinfo-types}
+\end{figure*}
+
+\begin{figure*}[htb]
+  \begin{align*}
+    & \emph{Script Purpose Types} \\
+    & \mathsf{Cert} & \type{P.Certifying} \\
+    & \mathsf{Rwrd} & \type{P.Rewarding} \\
+    & \mathsf{Mint} & \type{P.Minting} \\
+    & \mathsf{Spend} & \type{P.Spending} \\
+    ~\\
+    & \emph{Execution Budget Types} \\
+    & \Integer & \type{P.ExCPU} \\
+    & \Integer  & \type{P.ExMemory}\\
+    & \ExUnits & \type{P.ExBudget}  \\
+    ~\\
+    & \emph{TxInfo} \\
+    & \TxInfo = \\
```

Comment:
This definition is very difficult to read. If the comments don't fit in the same line, they should be shortened until they do, with extra information given in the prose. For example, the comment for the mint line should just be `mint` or `mint field`.

Reply:
the description of the translation is not trivial - it needs to be mentioned in the comment (i dont have a better idea of where to put it)